scripts/validate_oas_examples.py

```diff
@@ -157,44 +165,47 @@ def validate_request_examples():
     Validates examples for requests
     """
     # Build request validator
-    req_validator = RequestValidator(spec)
 
     # Build data object with endpoints and examples
     endpoints_and_examples_request = build_request_validation_data()
 
+    endpoint_count = 0
+    file_count = 0
     # Validate request examples
     for endpoint in endpoints_and_examples_request:
+        endpoint_count += 1
         for example_req_path in endpoints_and_examples_request[endpoint]:
-
-            abs_path = os.path.join(REPO_ROOT, "specification", example_req_path)
+            file_count += 1
+            example_path = (
+                "components/r4/" if endpoint.startswith("/R4/") else "components/stu3/"
+            )
+            abs_path = os.path.join(
+                REPO_ROOT,
+                "specification",
+                example_req_path.replace("../../", example_path),
+            )
             with open(abs_path, "r") as example_file:
                 example_request = load(example_file)
 
-            oapi_req = OpenAPIRequest(
-                full_url_pattern=base_url + endpoint,
-                method=get_http_method(endpoint),
-                body=example_request,
-                mimetype="application/fhir+json",
+            request = Request(
+                get_http_method(endpoint),
+                base_url + endpoint,
+                headers={"content-type": "application/fhir+json"},
+                json=example_request,
             )
+            oapi_req = RequestsOpenAPIRequest(request)
 
             # Validate request body
-            result = req_validator._validate_body(oapi_req)
-
-            # Raise errors if request invalid
-            try:
-                result.raise_for_errors()
-            except Exception as e:
-                print("\nError: JSON data file with path " + abs_path)
-                print(
-                    "used in request example for endpoint "
-                    + endpoint
-                    + " is not valid."
-                )
-                print("See 'ValidationError' field below:")
-                print(e)
-                exit(1)
```

Comment:
If an error is thrown, should the script fail with some kind of message? Not sure if the library will stop the script on error or not.